diff --git a/org.claws_mail.Claws-Mail.desktop b/org.claws_mail.Claws-Mail.desktop
index 75f20bb..4f038af 100644
--- a/org.claws_mail.Claws-Mail.desktop
+++ b/org.claws_mail.Claws-Mail.desktop
@@ -20,7 +20,7 @@ GenericName[sk]=Poštový klient
 GenericName[sv]=E-postklient
 GenericName[tr]=E-posta istemcisi
 Exec=claws-mail %u
-Icon=claws-mail
+Icon=org.claws_mail.Claws-Mail
 Categories=Network;Email;
 Keywords=lightweight;fast;gui;extensible;plugin;pop;pop3;imap;imap4;nntp;news;
 Comment=Lightweight and Fast GTK+ based Mail Client
